### PR TITLE
Ft secrets injection

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
**_WHAT DOES THIS PR DO?_**
- Automates the configmap update process for responding deployments and cronjobs

**_WHAT ISSUES ARE RELATED TO THIS PR?_**
- Jira cards
    - [OPS-39](https://airqoteam.atlassian.net/browse/OPS-39?atlOrigin=eyJpIjoiNGI5NzE3YmUzZGYxNDMwODg5NTk1NWVhMDU1NGQyYmUiLCJwIjoiaiJ9)

**_HOW DO I TEST OUT THIS PR?_**
- You can mainly do static code analysis for these changes
**Or you can:**
- Fork this repository and switch to this branch. 
  `git clone https://github.com/airqo-platform/AirQo-api.git`
  `git checkout ft-secretsInjection`
- Add a version to any staging environment secret using the secrets generic accessor account on the GCP Secret Manager.
- Make a change in the respective microservice for which you have just added a secret version
- Commit your changes and create a PR against the staging branch of your fork
- Merge the PR
- After the actions have completed running, checkout the configmap of the microservice you updated
- You should see the changes you made reflecting in that configmap

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
- n/a

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
- n/a

**_ARE THERE ANY RELATED PRs?_**
- [This PR](https://github.com/airqo-platform/AirQo-frontend/pull/765) in the frontend repository

### CODE REVIEW

File: .github/workflows/deploy-apis-to-production.yml [hunk 1]
```diff
@@ -242,14 +243,31 @@ jobs:
...
+        with:
+          credentials_json: ${{ secrets.GCP_SA_CREDENTIALS }}
+
+      - name: Setup Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0
+
+      - name: Update the corresponding k8s configmap(s)
+        run: |
+          cd src/auth-service/
+          gcloud secrets versions access latest --secret="prod-env-auth-service" > .env
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:266]: Downloading secret and saving it into a .env file

File: .github/workflows/deploy-apis-to-production.yml [hunk 2]
```diff
@@ -242,14 +243,31 @@ jobs:
...
+          kubectl create configmap --dry-run=client -o yaml --from-env-file=.env prod-auth-api-config | kubectl replace -f - -n production
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:267]: Updating the configmap on Kubernetes

Comment (NoahNsimbe) [reply .github/workflows/deploy-apis-to-production.yml:267]: Hi @123MwanjeMike are you sure you want to overwrite??...Why not just update configmaps?

Comment (123MwanjeMike) [reply .github/workflows/deploy-apis-to-production.yml:267]: Yes @NoahNsimbe 
That step simply updates the configmaps.
A copy of the actual value is maintained in Secret Manager as our source of truth

Comment (NoahNsimbe) [reply .github/workflows/deploy-apis-to-production.yml:267]: Cool

**Review by NoahNsimbe — APPROVED** [review]
LGTM. Thanks @123MwanjeMike

**Review by Baalmart — APPROVED** [review]
LGTM, thanks @123MwanjeMike

**Pull request merged**